controllers/imagejob/imagejob_controller.go

```diff
@@ -82,17 +95,69 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// your logic here
+	controllerLog.Info("imagejob reconcile")
 
-	// create eraser pod in every node like broadcast job does
-	/// get list of nodes and schdules the pods on each of the nodes using pod.spec.nodename
-	// eventually check if pod fits before doing that
-
-	//var imageJob eraserv1alpha1.ImageJob
-	//if err := r.Get
+	nodes := &v1.NodeList{}
+	err := r.List(context.TODO(), nodes)
+	if err != nil {
+		panic(err)
+	}
 
+	count := 0
+
+	// only get names from for loop
+	for _, n := range nodes.Items {
+		count++
+		controllerLog.Info("inside nodes.Items for loop")
+		nodeName := n.Name
+
+		runTime := n.Status.NodeInfo.ContainerRuntimeVersion
+		runTimeName := strings.Split(runTime, ":")[0]
+
+		var socketPath string
+
+		if runTimeName == "dockershim" {
+			socketPath = "/var/run/dockershim.sock"
+		} else if runTimeName == "containerd" {
+			socketPath = "/run/containerd/containerd.sock"
+		} else if runTimeName == "crio" {
```

Comment:
this should be `cri-o`

Reply:
looks like this wasn't resolved?

Reply:
we did the changes in eraser refactoring instead, they should be pulled in now